docs/src/main/asciidoc/rest-client-reactive.adoc

```diff
@@ -0,0 +1,594 @@
+
+    @RestClient
+    CountriesService countriesService;
+
+
+    @GET
+    @Path("/name/{name}")
+    @Blocking
+    public Set<Country> name(@PathParam("name") String name) {
+        return countriesService.getByName(name);
+    }
+
+    @GET
+    @Path("/name-async/{name}")
+    public CompletionStage<Set<Country>> nameAsync(@PathParam("name") String name) {
+        return countriesService.getByNameAsync(name);
+    }
+}
+----
+
+Please note that since the invocation is now non-blocking, we don't need the `@Blocking` annotation anymore on the endpoint.
+This means that the `nameAsync` method will be invoked on the event loop, i.e. will not get offloaded to a worker pool thread.
+
+
+To test asynchronous methods, add the test method below in `CountriesResourceTest`:
+[source,java]
+----
+@Test
+public void testCountryNameAsyncEndpoint() {
+    given()
+    .when().get("/country/name-async/greece")
+    .then()
+        .statusCode(200)
+        .body("$.size()", is(1),
+             "[0].alpha2Code", is("GR"),
+             "[0].capital", is("Athens"),
+             "[0].currencies.size()", is(1),
+             "[0].currencies[0].name", is("Euro")
+        );
+}
+----
+
+The `Uni` version is very similar:
+
+[source, java]
+----
+package org.acme.rest.client;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+@Path("/v2")
+@RegisterRestClient
+public interface CountriesService {
+
+    // ...
+
+    @GET
+    @Path("/name/{name}")
+    @Produces("application/json")
+    Uni<Set<Country>> getByNameAsUni(@PathParam("name") String name);
+}
+----
+
+The `CountriesResource` becomes:
+
+[source,java]
+----
+package org.acme.rest.client;
+
+import io.smallrye.common.annotation.Blocking;
+import io.smallrye.mutiny.Uni;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+@Path("/country")
+public class CountriesResource {
+
+    @RestClient
+    CountriesService countriesService;
+
+
+    // ...
+
+    @GET
+    @Path("/name-uni/{name}")
+    public Uni<Set<Country>> nameUni(@PathParam("name") String name) {
+        return countriesService.getByNameAsUni(name);
+    }
+}
+----
+
+[TIP]
+.Mutiny
+====
+The previous snippet uses Mutiny reactive types, if you're not familiar with them, read the link:getting-started-reactive#mutiny[Getting Started with Reactive guide] first.
+====
+
+When returning a `Uni`, every _subscription_ invokes the remote service.
+It means you can re-send the request by re-subscribing on the `Uni`, or use a `retry` as follows:
+
+[source, java]
+----
+
+@Inject @RestClient CountriesResource service;
```

Comment:
`@Inject` isn't needed here as you have mentioned above :)

Reply:
👍🏻